graph_dfs_debug/graph.py

```diff
@@ -16,25 +16,24 @@ def __init__(self):
         self.components = 0
 
     def add_vertex(self, vertex, edges=()):
-        self.vertices[vertex] = set(edges)
+      self.vertices[vertex] = set(edges)
 
     def add_edge(self, start, end, bidirectional=True):
-        self.vertices[start].add(start)
-        if bidirectional:
-            self.vertices[end].add(end)
+      self.vertices[start].add(end) #start to end
+      if bidirectional:
+         self.vertices[end].add(start) #end to start
 
     def dfs(self, start, target=None):
         x = []
-        x.append(start)
         y = set(x)
-
+        x.append(start)
         while x:
-            z = x.pop()
-            if x == target:
-                break
-            x.extend(self.vertices[z])
-
-        return x
+          z = x.pop()
+          if z == target:
+            break
+          y.add(z) # add x to 
+          x.extend(self.vertices[z]-y)
+        return y
```

Comment:
consider naming variables with more intuitive names..reading this..I have no idea what z and y are..and why they're doing what they're doing